DMTN-306.tex

```diff
@@ -1,79 +1,121 @@
+
+\subsection{Registration of files to replicate}
+\label{registration}
+
+To perform replication, we create Rucio Datasets, each composed of a set of files that are already in their appropriate locations at the source RSE. Upon registration, pre-configured Rucio subscriptions trigger the actual file movement to the destination facility, in accordance with the defined replication rules. Rucio delegates the data transfer to FTS, which then instructs the storage endpoints at the facilities to move the data, typically by requesting the destination facility to pull the data from the source facility. The use of Datasets allows grouping of related files, use of subscription patterns applied to spatially-defined Dataset names to associate spatial regions with RSEs, and a clear way to know when all related files have been generated (via Dataset closure) and replicated.
+
+Rubin has developed the tool \texttt{rucio\_register}\footnote{https://github.com/lsst/rucio\_register}, which allows for the selection of existing files from a Butler repository based on specified criteria. The tool attaches Rubin-specific metadata to these files and registers them into one or more Rucio Datasets. The metadata, encoded as a JSON record, contains a minimal set of information extracted from the origin Butler repository. This ensures that replicated files can be ingested properly into the local Butler repository at the destination facility.
+
+Only files that require replication to another facility are registered with Rucio. As a result, Rubin's instance of Rucio is aware only of files replicated across processing facilities. Files that are local to each facility and not subject to replication remain known only to that facility's Butler and are not registered in Rucio. Since the US Data Facility gets a complete copy of all final data products, by definition files that are not replicated are intermediates in the calculations that are not required to be persisted.
+
+\subsection{Ingestion at reception}
+\label{ingestion}
+
+FTS notifies Rucio about the completion of individual file transfers. Rubin's \texttt{HermesK}\footnote{https://github.com/lsst-dm/ctrl\_rucio\_ingest}, which is a modification of Rucio's Hermes daemon, filters messages and uses Kafka as a mechanism to signal the destination Rubin facility that a new file was replicated and to take appropriate actions. Kafka was selected as a reliable message bus used for other purposes within the Rubin project. Its ordering guarantees are not strictly necessary in this application, but the ability to scale to multiple consumers may be needed as the number of files increases.
```

Comment:
after "within the Rubin project" maybe add an "e.g. Fausti Neto et al 2024; Ribeiro et al 2024" citation (https://ui.adsabs.harvard.edu/abs/2024SPIE13101E..1MF/abstract and https://ui.adsabs.harvard.edu/abs/2024SPIE13101E..18R/abstract).

Reply:
Use `\url` with the URL footnotes.

Reply:
@srp3rd can you tweak the ctrl_rucio_ingest repo at some point to add a repo to make it a bit more obvious why we call it HermesK here but then it's in a repo that doesn't mention Hermes at all?

Reply:
Add a "repo" or maybe you meant "README note"?

Reply:
I did mean README...